File: Day 74/328. Odd Even Linked List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:

        if(head is None):
            return None

        temp = head
        tempEven = head.next

        flag = True

        while (temp.next is not None):
            curr = temp.next
            temp.next = curr.next
            temp = curr
            flag = not flag
        

        if (flag):
            temp.next = tempEven
            return head
        

        else:
            temp = head
            while(temp.next is not None):
                temp = temp.next
            temp.next = tempEven
            return head
        

File: Day 74/328. Odd Even Linked List.cpp
#include <iostream>
using namespace std;

    
 struct ListNode {
    int val;
     ListNode *next;
     ListNode() : val(0), next(nullptr) {}
      ListNode(int x) : val(x), next(nullptr) {}
      ListNode(int x, ListNode *next) : val(x), next(next) {}
  };
 
class Solution {
public:
    ListNode* oddEvenList(ListNode* head) {

        if(head == nullptr) return nullptr;

        ListNode* temp = head;
        ListNode* tempEven = head -> next;

        bool flag = true;

        while(temp -> next != nullptr && temp != nullptr){
            ListNode* curr = temp -> next;
            temp -> next = curr -> next;
            temp = curr;
            flag = !flag;
        }

        if(flag){
            temp -> next = tempEven;
            return head;
        }

        else{
            temp = head;
            while(temp -> next != nullptr) temp = temp -> next;
            temp -> next = tempEven;
            return head;
        }
        
    }
};

int main(){
  
  return 0;
}